sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java

```diff
@@ -472,6 +483,28 @@ public ConnectionPolicy setPreferredRegions(List<String> preferredRegions) {
         return this;
     }
 
+    public ConnectionPolicy setExcludedRegions(List<String> excludedRegions) {
+        reentrantReadWriteLock.writeLock().lock();
+        try {
+            if (excludedRegions != null) {
+                this.excludedRegions = new ArrayList<>(excludedRegions);
+            }
+            this.excludedRegionsAsString = constructExcludedRegionsAsString(this.excludedRegions);
```

Comment:
If this is just used for diagnostics and `toString()` purposes, then we should remove this from the lock and this calculation (making it slow), rather lazily compute this when creating diagnostics or calling `toString()`, thoughts?

Reply:
Good point but I don't anticipate the setter to be called often whereas stringifying the `DiagnosticsClientConfig` for diagnostics happens for every operation and each time we would have to recompute `excludedRegionsAsString`.